src/api.cc

```diff
@@ -922,3 +1034,165 @@ InstallResult AkliteClient::SetSecondaries(const std::vector<SecondaryEcu>& ecus
+  do {
+    try {
+      ++version;
+      latest_root = bundle_tuf_repo.FetchRoot(version);
+      if (repo.rootVersion() > 0) {
+        repo.verifyRoot(latest_root);
+      } else {
+        repo.initRoot(Uptane::RepositoryType::Image(), latest_root);
+      }
+    } catch (const aklite::tuf::MetadataNotFoundException& exc) {
+      break;
+    }
+  } while (version < max_version);
+
+  if (latest_root.empty()) {
+    throw BundleMetaError(BundleMetaError::Type::RootMetaNotFound,
+                          "Failed to find root metadata; missing version: " + std::to_string(version));
+  }
+  return Utils::parseJSON(latest_root);
+}
+
+static Json::Value checkAndGetBundleMeta(const std::shared_ptr<aklite::tuf::Repo>& device_tuf_repo,
+                                         const boost::filesystem::path& bundle_tuf_repo_path) {
+  const auto root_meta{checkAndGetRootMeta(device_tuf_repo, bundle_tuf_repo_path)};
+
+  std::set<std::string> targets_key_ids;
+  for (const auto& key : root_meta["signed"]["roles"]["targets"]["keyids"]) {
+    targets_key_ids.insert(key.asString());
+  }
+
+  std::map<std::string, PublicKey> targets_pub_keys;
+  for (Json::ValueConstIterator it = root_meta["signed"]["keys"].begin(); it != root_meta["signed"]["keys"].end();
+       ++it) {
+    const std::string keyid{it.key().asString()};
+    if (targets_key_ids.count(keyid) == 1) {
+      targets_pub_keys.insert({keyid, PublicKey{*it}});
+    }
+  }
+
+  const int targets_sign_threshold{root_meta["signed"]["roles"]["targets"]["threshold"].asInt()};
+
+  int found_valid_sig_number{0};
+  const auto bundle_meta_path{bundle_tuf_repo_path / "bundle-targets.json"};
+  if (!boost::filesystem::exists(bundle_meta_path)) {
+    throw BundleMetaError(BundleMetaError::Type::NotFound,
+                          "Failed to find the bundle metadata: " + bundle_meta_path.string());
+  }
+  const auto bundle_targets_str{Utils::readFile(bundle_meta_path)};
+  auto bundle_targets_json{Utils::parseJSON(bundle_targets_str)};
+  if (bundle_targets_json.isNull() || bundle_targets_json.empty() || !bundle_targets_json.isMember("signed") ||
+      !bundle_targets_json.isMember("signatures")) {
+    throw BundleMetaError(BundleMetaError::Type::InvalidMetadata,
+                          "The bundle metadata is invalid;"
+                          " expecting JSON with `signed` and `signatures` fields; got: \n" +
+                              bundle_targets_str);
+  }
+  if (!bundle_targets_json["signatures"].isArray()) {
+    throw BundleMetaError(BundleMetaError::Type::InvalidMetadata,
+                          "The bundle metadata is invalid;"
+                          " the `signatures` field is not an array; got: \n" +
+                              bundle_targets_str);
+  }
+
+  const std::string signed_body{Utils::jsonToCanonicalStr(bundle_targets_json["signed"])};
+  LOG_INFO << "Checking the bundle metadata singatures (required: " << targets_sign_threshold << ")...";
```

Comment:
```suggestion
  LOG_INFO << "Checking the bundle metadata signatures (required: " << targets_sign_threshold << ")...";
```

Reply:
Fixed in the force push.

(I turned off the spell checker in my IDE because it eats too much CPU, it looks like I have to find another spell checker :)).